plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -42,4 +48,41 @@ public static boolean isManagedSdkFeatureEnabled() {
     }
     return false;
   }
+
+  /**
+   * Installs a Cloud SDK, if the preferences are configured to auto-manage the SDK. Blocks callers
+   * 1) if the managed SDK is being installed concurrently by others; and 2) until the installation
+   * is complete.
+   *
+   * @param consoleStream stream to which the install output is written
+   */
+  public static void installManagedSdk(MessageConsoleStream consoleStream)
```

Comment:
Do you think just using `OutputStream` might be better?

Reply:
We have `MessageConsoleWriterListener` which accepts the `MessageConsoleStream`. The other day, I thought the same thing and tried to use `OutputStream` but realized it is a bit cumbersome and needs more code as the native `OutputStream` doesn't have `println`.